addons/hr_recruitment/wizard/applicant_refuse_reason.py

```diff
@@ -67,40 +67,36 @@ def action_refuse_reason_apply(self):
         if self.send_mail:
             if not self.template_id:
                 raise UserError(_("Email template must be selected to send a mail"))
-            if not self.applicant_ids.filtered(lambda x: x.email_from or x.partner_id.email):
-                raise UserError(_("Email of the applicant is not set, email won't be sent."))
+            if any(not (applicant.email_from or applicant.partner_id.email) for applicant in self.applicant_ids):
+                raise UserError(_("At least one applicant doesn't have a email; you can't use send email option."))
 
         refused_applications = self.applicant_ids
+        # duplicates_count can be true only if only one application is selected
         if self.duplicates_count and self.duplicates:
-            duplicate_domain = self.applicant_ids._get_similar_applicants_domain()
-            duplicate_domain = expression.AND([duplicate_domain, [('application_status', '!=', 'refused')]])
-            duplicates_ids = self.env['hr.applicant'].search(duplicate_domain)
+            applicant_id = self.applicant_ids[0]
+            duplicate_domain = applicant_id.candidate_id._get_similar_candidates_domain()
+            duplicates_ids = self.env['hr.candidate'].search(duplicate_domain).applicant_ids
```

Comment:
Those are records, not ids :)
```suggestion
            duplicates = self.env['hr.candidate'].search(duplicate_domain).applicant_ids
```